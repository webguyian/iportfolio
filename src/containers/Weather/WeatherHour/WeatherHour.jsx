import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon/Icon';
import Text from 'components/Text/Text';

import { getHour, formatTemp } from 'modules/weather/helpers';

const WeatherHour = props => {
  const { icon, temperature, time } = props;
  const baseClass = 'weather-hour';

  return (
    <li className={baseClass}>
      <Text className={`${baseClass}-time`}>{getHour(time)}</Text>
      <Icon svg className={`${baseClass}-icon`} name={icon} />
      <Text className={`${baseClass}-temperature`}>
        {formatTemp(temperature)}
      </Text>
    </li>
  );
};

WeatherHour.propTypes = {
  icon: PropTypes.string,
  temperature: PropTypes.number,
  time: PropTypes.number
};

export default WeatherHour;
