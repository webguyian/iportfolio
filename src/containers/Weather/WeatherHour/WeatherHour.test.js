import React from 'react';
import { create } from 'react-test-renderer';

import WeatherHour from './WeatherHour';

describe('<WeatherHour />', () => {
  const props = {
    icon: 'clear-day',
    temprature: 69.41,
    time: 1583726400
  };

  it('renders correctly', () => {
    const component = create(<WeatherHour {...props} />);

    expect(component).toMatchSnapshot();
  });
});
