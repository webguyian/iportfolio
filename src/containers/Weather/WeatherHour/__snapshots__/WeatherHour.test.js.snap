// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherHour /> renders correctly 1`] = `
<li
  className="weather-hour"
>
  <span
    className="ui-text ui-text--body weather-hour-time"
  >
    12AM
  </span>
  <svg
    className="ui-icon ui-icon-clear-day weather-hour-icon"
    focusable="false"
    role="presentation"
  >
    <use
      height="100%"
      width="100%"
      xlinkHref="#clear-day"
    />
  </svg>
  <span
    className="ui-text ui-text--body weather-hour-temperature"
  >
    NaNÂ°
  </span>
</li>
`;
